metaseq/cli/train.py

```diff
@@ -308,9 +317,18 @@ def train(
 
         return valid_losses, should_stop
 
+    dcfg = DynamicConfig(
+        json_file_path=cfg.common.dynamic_config_path,
+        timeout=cfg.common.dynamic_config_timeout,
+    )
+
     for i, samples in enumerate(progress):
-        if distributed_utils.get_global_rank() == 0 and cfg.common.profile and i == 5:
-            logger.info("STARTING PROFILER")
+        force_profile = dcfg["force_profile"]
+        do_profile = (distributed_utils.get_global_rank() == 0) and (
+            (cfg.common.profile and i == 5) or force_profile
```

Comment:
Why `i == 5`?

Should `cfg.common.profile` still be a way to enable tracing, or should we just switch to the dynamic configs?

Reply:
I am trying to make this backward-compatible so that I don't break anyone else's classical way of doing things. Ideally, all of the logging/profiling work should be controlled by dynamic configuration, but that might be unusual for the rest of the team if included in the very first PR

Reply:
I like that this doesn't break others' way of doing things.

I am still concerned about `i == 5` :)

Reply:
i == 5 is an arbitrary choice of @ngoyal2707